### DISCUSSION TITLE
Project not longer opensource because of PR: #17010

### DISCUSSION THREAD

**Discussion by stbischof** [eslint/eslint#17175] [General] [answered]
With https://github.com/eslint/eslint/pull/17010 / https://github.com/eslint/eslint/issues/16980 a NON-OpenSource License was added to OpenSource/MIT Code:

1. Since this Project is specified to have default MIT License [here](https://github.com/eslint/eslint/blob/main/LICENSE) and [here](https://camo.githubusercontent.com/ca0a233b5442fba44471d76afe67ffb991fd317428a75c36cf944b1de247f425/68747470733a2f2f6170702e666f7373612e696f2f6170692f70726f6a656374732f67697425324268747470732533412532462532466769746875622e636f6d25324665736c696e7425324665736c696e742e7376673f747970653d6c61726765)  your argument seems  wrong:

```
Currently we don't have a documentation-specific license, so it defaults to all rights reserved. 
```

2. MIT - license covers `and associated documentation`
so the PR is braking the license of source

```
Permission is hereby granted, free of charge, to any person obtaining a copy
of this software and associated documentation files (the "Software"),
```

3. Public founded Projects in Germany are now very often having open Source Rules:

Since [Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International License](https://creativecommons.org/licenses/by-nc-sa/4.0/) is not compatible to OpenSource  - [see Rules for Open Source in public founded projects](https://wikijs.opencode.de/de/Hilfestellungen_und_Richtlinien/Lizenzcompliance) .

They are not allowed to use/work on this projects with this public money. Since a Developer must write not only code but also docs its now impossible to contribute.

**Comment by nzakas** [comment]
Thanks for bringing this up. I'm not sure if what you're saying is true since the only reference you cited is written in German and I don't know German. 

> They are not allowed to use/work on this projects with this public money. Since a Developer must write not only code but also docs its now impossible to contribute.

Who is "they"?

In any event, it's incorrect to say that the project is no longer open source because of a CC license applied to the docs. Other projects, such as [libuv](https://github.com/libuv/libuv), do similar things without problem.

I'm not opposed to removing the CC license, as it does seem that MIT covers licensing the documentation as well, but I would like some more background information.

**Reply by stbischof** [reply]
So we have basicly 2 issues:

1. MIT license covers the Docs. And you added a additional license. Questions here are:
- why adding an additional license?
- is is okay to do that because license is comoatible and all committers agree.

2. Wich effect does this license have to other projects. Options:
- no effects because license is still MIT. Adding later an other license has no legal effects.
- using an other license breaks the old license. That means you could do the same with the code. Because code and docs are named both in MIT license Code.
- new code would could not be MIT license when author must write docs

**Reply by rahulmohang** [reply]
Hi @nzakas, I understand that it is fine to license the documentation under the CC license even if the main license is MIT. The problem here is that when the license of the documentation is a non commercial license, some open source projects like https://projects.eclipse.org/projects/automotive.tractusx, that uses eslint, may have to violate our policy of not including files licensed under non commercial licenses in our projects.

**Reply by stbischof** [reply]
[libuv](https://github.com/libuv/libuv) uses:
Creative Commons Attribution 4.0 International | CC-BY-4.0

you:
Creative Commons Attribution Non Commercial 4.0 International | CC-BY-NC-4.0

If you had chosen `CC-BY-4.0` over ´CC-BY-NC-4.0` no License Check Systems had alerted me. 

For all contributors that made commits to the docs it might be still surprising that you re-license from MIT to....

**Comment by stbischof** [comment]
The Main issue for my project is that we are allowed to use just open source Software that is non restriktiv..your moved to a non commercial license. Which we are not allowed to use.


[Listing of Non OpenSource](Licenseshttps://wikijs.opencode.de/de/Hilfestellungen_und_Richtlinien/Lizenzcompliance#h-3-negativliste-aller-nicht-freigegebenen-lizenzen)
If you are interested in concret issues of our project with license i can write you by mail.

**Comment by stbischof** [comment]
@nzakas Could we help with a PR that reverts the license back to MIT, or an non NON COMMERCIAL (NC) license. like `CC-BY-4.0`

**Comment by nzakas** [comment, marked as answer]
Sorry, I've been moving and have not been online for most of the past two weeks.

I spoke with the OpenJS Foundation legal team about this. To summarize:

1. The project is still considered open source in its current state. Having a different license for documentation doesn't negate the MIT license for the code.
2. The only issue is if someone is committing both documentation and code at the same time.
3. MIT did cover documentation originally, but we didn't understand that, which is why the CC license was chosen.
4. We can fix this by either reverting back to the MIT license or switching to a CC-BY 4.0 license.

So the action here is to determine which license to use, and as a followup, we'll need to find all of the documentation commits after the license was introduced and pull those out, update the license, then ask the original authors to resubmit.

Here's the issue to track the work:
https://github.com/eslint/eslint/issues/17225

**Reply by stbischof** [reply]
I think that moving to CC-BY 4.0 is a really good . Thx
